source/plugin/offline-userplayer-data.rst

```diff
@@ -0,0 +1,71 @@
+===================
+Offline Player Data
+===================
+
+It may be necessary for plugins to access player data even when the player is offline.
+One could guess that ``Sponge.getServer().getPlayer()`` returning a ``Player`` can be used for this.
+But since ``Player`` objects only exist for online players, another solution has to be used.
+
+Some plugins store the relevant data themselves and associate the user by using the ``GameProfileManager``.
+But writing different code for offline and online users is not necessary:
+The ``ServiceManager`` natively provides a service known as the ``UserStorageService`` which is capable of returning ``User`` instances for ``Player``\s who are currently offline.
+Since the ``Player`` interface extends ``User`` most methods you call on a ``Player`` are also available. 
+
+For example:
+
+* ``#hasPermission(String permission)`` is available from both instances.
+
+Code Example
+----------------------------------
+
+Here's an example for a utility method that can be used to get a User:
+
+.. code-block:: java
+
+    //(Only sponge imports listed)
+    import org.spongepowered.api.Sponge;
+    import org.spongepowered.api.entity.living.player.User;
+    import org.spongepowered.api.service.user.UserStorageService;
+    
+    /**
+     * A method that can return offline and online users
+     * @return Optional<User> User reference. Non present means unknown
+     */
```

Comment:
hmm, I get what you mean, but IMO comments make this code snippet more understandable.
It might also give ppl a little poke so they document their code too.